server/src/main/java/org/elasticsearch/index/query/MatchNoneQueryBuilder.java

```diff
@@ -25,15 +25,28 @@
 public class MatchNoneQueryBuilder extends AbstractQueryBuilder<MatchNoneQueryBuilder> {
     public static final String NAME = "match_none";
 
+    private QueryBuilder originalQueryBuilder;
```

Comment:
Whatever field we add here we need to include in the query builder serialization (look at other query builders constructors that take a StreamInput and the "doWriteTo" method that is used to write query builder properties when they are sent to other nodes.
This needs to happen in a backwards compatible manner, e.g. look at https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/query/MatchQueryBuilder.java#L120 how this is usually done)

Reply:
I will have a look. Thanks for pointing this out.